python/cudf/cudf/io/orc.py

```diff
@@ -183,34 +188,46 @@ def _filter_stripes(
         filepath_or_buffer, columns_in_predicate
     )
 
-    # Filter using file-level statistics
-    if not ioutils._apply_filters(filters, file_statistics):
-        return []
-
-    # Filter using stripe-level statistics
-    selected_stripes = []
-    num_rows_scanned = 0
-    for i, stripe_statistics in enumerate(stripes_statistics):
-        num_rows_before_stripe = num_rows_scanned
-        num_rows_scanned += next(iter(stripe_statistics.values()))[
-            "number_of_values"
-        ]
-        if stripes is not None and i not in stripes:
-            continue
-        if skip_rows is not None and num_rows_scanned <= skip_rows:
+    file_stripe_map = []
+    for file_stat in file_statistics:
+        # Filter using file-level statistics
+        if not ioutils._apply_filters(filters, file_stat):
             continue
-        else:
-            skip_rows = 0
-        if (
-            skip_rows is not None
-            and num_rows is not None
-            and num_rows_before_stripe >= skip_rows + num_rows
-        ):
-            continue
-        if ioutils._apply_filters(filters, stripe_statistics):
-            selected_stripes.append(i)
 
-    return selected_stripes
+        # Filter using stripe-level statistics
+        selected_stripes = []
+        num_rows_scanned = 0
+        for i, stripe_statistics in enumerate(stripes_statistics):
+            num_rows_before_stripe = num_rows_scanned
+            num_rows_scanned += next(iter(stripe_statistics.values()))[
+                "number_of_values"
+            ]
+            if stripes is not None and i not in stripes:
+                continue
+            if skip_rows is not None and num_rows_scanned <= skip_rows:
+                continue
+            else:
+                skip_rows = 0
+            if (
+                skip_rows is not None
+                and num_rows is not None
+                and num_rows_before_stripe >= skip_rows + num_rows
+            ):
+                continue
+            if ioutils._apply_filters(filters, stripe_statistics):
+                selected_stripes.append(i)
+
+        file_stripe_map.append(selected_stripes)
+
+    return file_stripe_map
+
+
+def _ensure_filesystem(passed_filesystem, path):
```

Comment:
Can we add a small doc for this function?

Reply:
You know what. Really this function is used in almost all of the ioreaders. Should I just move it to ioutils.py and make it common across all classes?

Reply:
Could be future optimization. But good for this PR for now.